cpp/test/dist_sort_test.cpp

```diff
@@ -0,0 +1,98 @@
+/*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "common/test_header.hpp"
+#include "test_utils.hpp"
+#include <cylon/table.hpp>
+
+namespace cylon {
+namespace test {
+TEST_CASE("Dist sort testing", "[dist sort]") {
+    std::string path1 = "../data/input/csv1_" + std::to_string(RANK) + ".csv";
```

Comment:
Rather than using files, this is an alternative method to generate a global table within a test case and slice it based on the rank. 
https://github.com/cylondata/cylon/blob/b2c0820744e148b807da3f44658bb91c64cc8271/cpp/test/aggregate_test.cpp#L319-L409

These tests could be done in the following way (This removes the dependency on INITIAL_SAMPLE sorting method for tests) 
1. Slice global table to create each partition for ranks
2. Do distributed sort
3. At rank 0, locally sort the global table using `arrow::compute::SortIndices` + `arrow::compute::Take`. Other ranks would have an empty table with the same schema. 
4. Call ordered `DistributedEquals` on globally sorted table created by step 3 and distributed sort result